File: api/src/main/java/net/blueshell/api/repository/FileRepository.java
package net.blueshell.api.repository;

import net.blueshell.api.base.BaseRepository;
import net.blueshell.api.model.File;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface FileRepository extends BaseRepository<File> {
    Optional<File> findByName(String name);
    Optional<File> findByEventBannerId(Long bannerId);
}

File: api/src/main/kotlin/net/blueshell/api/repository/FileRepository.kt
package net.blueshell.api.repository

import net.blueshell.api.repository.base.BaseRepository
import net.blueshell.api.model.File
import org.springframework.stereotype.Repository
import java.util.*

@Suppress("FunctionName")
@Repository
interface FileRepository : BaseRepository<File, Long> {
    fun findByName(name: String): Optional<File>

    fun findFirstBy_eventBanners_Id_EventId(eventId: Long): Optional<File>

    fun findByPath(path: String): Optional<File>
}
